### DISCUSSION TITLE
rpi camera support for non rasbian (OpenWRT) on rpi

### DISCUSSION THREAD

**Discussion by t3therdev** [bluenviron/mediamtx#3297] [Questions]
## Which version are you using?

v0.16.0

## Which operating system are you using?

- [x] Other (please describe) - Custom OpenWRT Build running on Rasberry Pi. 

## Describe the issue

Trying to use rpi camera support. Getting an error as its not rasbian: 
"[RPI Camera source] server was compiled without support for the Raspberry Pi Camera"

Our OS build is OpenWRT running on Rasberry PI and it has libcamera and libfreetype. 
Also, the rasberry pi camera is successfully mounted and can be used via v4l2. 

Can we manually specify the build of rpicamera in go or what would be required to try to enable this to get it to work?

**Comment by aler9** [comment]
Hello, if you are not using Raspbian, but you have all required dependencies, you can try to build the server from source by following instructions here:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#raspberry-pi

The "server was compiled without support" will disappear if you use the `-tags rpicamera` flag with `go build`.

[a comment by an automated account is not shown]
